### PR TITLE
CLOUDP-108667: Get all the private endpoints for the Data Lake in the specified project.

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-108667

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments
```bash
./bin/mongocli atlas privateendpoint dataLakes aws ls
ID                       ENDPOINT PROVIDER   TYPE        COMMENT
vpce-0fcd9d80bbafe1607   AWS                 DATA_LAKE   test mongocli
```
```bash
~/workspace/mongocli git:(CLOUDP-108667) ✗ ./bin/mongocli atlas privateendpoint dataLakes aws ls -o json
{
  "links": [
    {
      "rel": "self",
      "href": "https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/5e4e593f70dfbf1010295836/privateNetworkSettings/endpointIds?pageNum=1\u0026itemsPerPage=100"
    }
  ],
  "results": [
    {
      "endpointId": "vpce-0fcd9d80bbafe1607",
      "provider": "AWS",
      "type": "DATA_LAKE"
    }
  ],
  "totalCount": 1
}
```

### CODE REVIEW

**Review by gssbzn — DISMISSED** [review]
LGTM just don't forget to plan for e2e testing

File: go.mod [hunk 1]
```diff
@@ -21,7 +21,7 @@ require (
 	github.com/spf13/viper v1.10.1
 	github.com/stretchr/testify v1.7.0
 	github.com/tangzero/inflector v1.0.0
-	go.mongodb.org/atlas v0.14.0
+	go.mongodb.org/atlas v0.14.1-0.20220104180936-6afea6036494
```
Comment (drinkbird) [inline go.mod:24]: Out of curiosity, what would be the difference between `v0.14.1-0.20220104180936-6afea6036494` and `v0.14.1`?

Comment (andreaangiolillo) [reply go.mod:24]: if you go here [MongoDB Atlas Client Release v0.14.0](https://github.com/mongodb/go-client-mongodb-atlas/releases/tag/v0.14.0), you see that this release doesn’t have 4 commits from master
Screenshot 2022-01-12 at 15 55 25. In particular, the one that we are using here is https://github.com/mongodb/go-client-mongodb-atlas/pull/271

Comment (drinkbird) [reply go.mod:24]: Thanks, so it's a kind of pre-release for `v0.14.1`. My comment is non-blocking btw.

File: internal/cli/atlas/privateendpoints/datalake/aws/list.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	return opts.Print(r)
+}
+
+// mongocli atlas privateEndpoint(s)|privateendpoint(s) dataLakes aws list|ls [--projectId projectId].
+func ListBuilder() *cobra.Command {
+	opts := new(ListOpts)
+	cmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "List Data Lake Private Endpoints for your project.",
```
Comment (melissamahoney-mongodb) [inline internal/cli/atlas/privateendpoints/datalake/aws/list.go:63]: Just a suggestion to add a little more info:
```suggestion
		Short:   "List Data Lake private endpoints for your project.",
		Long:    "Your API key must have the GROUP_READ_ONLY (Project Data Access Read Only) or higher role to list private endpoints."
```

File: internal/cli/atlas/privateendpoints/datalake/aws/aws.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"github.com/spf13/cobra"
+)
+
+func Builder() *cobra.Command {
+	const use = "aws"
+	cmd := &cobra.Command{
+		Use:   use,
+		Short: "Manage AWS Private Endpoints.",
```
Comment (melissamahoney-mongodb) [inline internal/cli/atlas/privateendpoints/datalake/aws/aws.go:25]: It looks like we don't capitalize private endpoints in the other docs.
```suggestion
		Short: "Manage AWS private endpoints.",
```

File: internal/cli/atlas/privateendpoints/datalake/data_lake.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/mongodb/mongocli/internal/cli/atlas/privateendpoints/datalake/aws"
+	"github.com/spf13/cobra"
+)
+
+func Builder() *cobra.Command {
+	const use = "dataLakes"
+	cmd := &cobra.Command{
+		Use:     use,
+		Aliases: cli.GenerateAliases(use),
+		Short:   "Manage Data Lake Private Endpoints.",
```
Comment (melissamahoney-mongodb) [inline internal/cli/atlas/privateendpoints/datalake/data_lake.go:28]:
```suggestion
		Short:   "Manage Data Lake private endpoints.",
```

**Review by melissamahoney-mongodb — DISMISSED** [review]
A few small changes :)

**Review by gssbzn — APPROVED** [review]

File: internal/cli/atlas/privateendpoints/datalake/aws/list.go (hunk 2 above)
Comment (melissamahoney-mongodb) [reply internal/cli/atlas/privateendpoints/datalake/aws/list.go]: I don't think this change made it in (the additional Long description).

**Pull request merged**